packages/config-loader/api-report.md

```diff
@@ -45,6 +45,12 @@ export type ConfigSchema = {
   serialize(): JsonObject;
 };
 
+// @public
+export type ConfigSchemaPackageEntry = {
+  value: JsonObject;
+  path: string;
+};
```

Comment:
The `path` here is really just there for more informative error messages when validating the schema. I think it'd be best to avoid exporting this type and then do validation upfront instead, although that might be tricky. Alternatively it would likely make sense to rename this to `context` or something like that instead.

Similarly, `value` isn't a particularly good name here imo, probably best to have that be `schema`, `jsonSchema`, or something like that instead.

Reply:
Should I then rename both `path` to `context` and `value` to `jsonSchema` and still export it ?

Reply:
Done in commit e896620f2c1ca8d0feff24a6cf0d0c880bd6809e

Reply:
Well, this breaks unit tests for a number of plugins: https://github.com/backstage/backstage/actions/runs/7610309427/job/20723424247?pr=21632

That's because they implicitly depend on the field names `path` and `value` when testing again expected serialized JSON Values.

Should I revert this change ?
Or create a dedicated type with `context` and `jsonSchema` fields to be used in this PR ?

Reply:
In addition to that, I find other places where the `value` field is referenced like: https://github.com/backstage/backstage/blob/master/packages/cli/src/commands/config/schema.ts#L35.

This makes me think that I should not rename those fields now.

Reply:
For now I reverted the commit mentioned above and kept the existing names in the exported type, knowing that the field names are expected to be known in a number of places throughout the codebase.

Reply:
In fact I just made the `ConfigSchemapackageEntry` non public anymore, and simply used a map of JsonObject indexed on strings (the context of the external schema, most likely a path).
That seems clearer and doesn't impact the existing naming of the existing fields `ConfigSchemapackageEntry`, no exposes it.

The change is done in commit 84fcbc49d6d444bcfb347886ed8ca33cb139954b